src/main/java/hudson/tasks/junit/util/LinearRegression.java

```diff
@@ -0,0 +1,177 @@
+/******************************************************************************
+ *  Compilation:  javac LinearRegression.java
+ *  Execution:    java  LinearRegression
+ *  Dependencies: none
+ *
+ *  Compute least squares solution to y = beta * x + alpha.
+ *  Simple linear regression.
+ *
+ ******************************************************************************/
+
+package hudson.tasks.junit.util;
+
+/**
+ *  The {@code LinearRegression} class performs a simple linear regression
+ *  on an set of <em>n</em> data points (<em>y<sub>i</sub></em>, <em>x<sub>i</sub></em>).
+ *  That is, it fits a straight line <em>y</em> = &alpha; + &beta; <em>x</em>,
+ *  (where <em>y</em> is the response variable, <em>x</em> is the predictor variable,
+ *  &alpha; is the <em>y-intercept</em>, and &beta; is the <em>slope</em>)
+ *  that minimizes the sum of squared residuals of the linear regression model.
+ *  It also computes associated statistics, including the coefficient of
+ *  determination <em>R</em><sup>2</sup> and the standard deviation of the
+ *  estimates for the slope and <em>y</em>-intercept.
+ *
+ *  @author Robert Sedgewick
+ *  @author Kevin Wayne
+ */
+public class LinearRegression {
```

Comment:
Is this used somewhere?

Reply:
See **private void createLinearTrend** in History.java